common/thorhelper/thorsoapcall.cpp

```diff
@@ -1291,6 +1296,32 @@ bool httpHeaderBlockContainsHeader(const char *httpheaders, const char *header)
         return true;
     return false;
 }
+
+bool getHTTPHeader(const char *httpheaders, const char *header, StringBuffer& value)
+{
+    if (!httpheaders || !*httpheaders || !header || !*header)
+        return false;
+
+    StringArray headerList;
+    headerList.appendListUniq(httpheaders, "\n");
```

Comment:
This seems a little expensive for just retrieving one header value.  Could you just strstr to the header you want and extract it if it exists?  If we started extracting more headers then parsing them into an IProperties or something might make sense.